text/0002-numeric-values.md

```diff
@@ -0,0 +1,485 @@
+// so it converts `x` to a u8.
+const y: u8 = @intCast(x);
+// the result loation (the argument) is u16, so `@intCast` must produce a u16,
+// so it converts `x` to a u16.
+callExpectsU16(@intCast(x))
+```
+
+A mechanism like this would introduce a lot of compiler complexity, and kinda covers a different set of uses than exactly what the `.to(T)` function does. It is another RFC's job to decide this.
+
+## Why not use `number`?
+
+If we assigned a type to all JS numbers, like [TS `number`](https://www.typescriptlang.org/docs/handbook/basic-types.html#number), this leads to ambiguity, primarily in if the intent is to use an integer or a floating point number.
+
+If a developer really wants to have `number` and Blight accepts a 'types as values'-like proposal, they could assign `f64` to a constant named `number`.
+
+## Why only single float type `f64`?
+
+In the primary compilation target, JavaScript, uses 64-bit floats for all numbers. It does not give access to other percisions (32-bit, 128-bit). It would be impossible to compile any other float type than `f64` without embedding a library for correct calculations. For that reason, Blight does not have any other floating point types.
+
+`f32` could be implemented as [Float32Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Float32Array), but it would likely be slower than `f64`, which would be misleading to the developer.
+
+An alternative to this, which was used in an earlier version of this document, was `float`. It is more keys to type, but it looks cleaner, especially if other float sizes will not be added.
+
+## Why use a function `.pow` for exponentiation instead of `**`?
+
+It is ambiguous with a potential `*` unary operator. do you parse `a**b` as:
+
+- `a` `**` `b`
+- `a` `*` `(` `*b` `)`
+
+## Floating point overflow
+
+Should `f64.inf + 1` be `f64.inf` or a crash. This RFC defines that it is `f64.inf`.
+
+# Prior art
+
+[prior-art]: #prior-art
+
+A large part of this RFC is adapted from how Zig's [Integer](https://ziglang.org/documentation/0.11.0/#toc-Integers) and [Float](https://ziglang.org/documentation/0.11.0/#Floats) types work.
+
+The pronounciation of the types is adapted from the C99 [stdint.h](https://en.wikipedia.org/wiki/C_data_types#stdint.h) (`uint32_t`, `int64_t`, etc).
+
+Properties on the type definitions themselves are adapted from Rust's [Integer's Associated Constants](https://doc.rust-lang.org/std/primitive.i32.html)
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+## Handling Overflow Errors
+
+A way to handle overflow errors should be given. Making these methods on the values seems like the path of least resistance, but it will lead to long code like `something.tryAdd(63)` or whatever.
+
+Maybe later when we figure out error handling in general we can prescribe a way to handle operator errors easily.
+
+Similarly, this should be the case for `.sqrt`/`.pow` and so on
```

Comment:
Stupid idea: what if all operations simply return an optional type if they are fallible? This could play in nicely with a future error handling proposal.

Reply:
either an optional type or an error type. regardless, there is no current mechanism for this, so it will be deferred until that

Reply:
I think we should postpone error nonsense until later. I think if this spec simply says "may be an error" or something like that, that would be great.

Reply:
sorry, was this not under the future handling.

it was not. i meant this to be a future thing. i dont intend on solving the problem now